docs/spec/errors/error.json

```diff
@@ -84,6 +84,19 @@
                         "handled": {
                             "type": ["boolean", "null"],
                             "description": "Indicator whether the error was caught somewhere in the code or not."
+                        },
+                        "cause": {
+                            "type": ["array", "null"],
+                            "items": {
+                                "type": ["object", "null"],
```

Comment:
i don't think jsonschema is made for recursive definitions, and as per having a sum type, i don't want to have more complications than necessary honestly

Reply:
Sounds fair to me, in this case I suggest to check the conversion boolean at https://github.com/elastic/apm-server/pull/2532/files/0e0256d45bf0a4e070aa8a7a44da6229c02d87c1#diff-52137911864c236a97b14516397a04c6R435 though, as the `cause` has only been defined to be an object. Not doing the check would lead to nil values, just dropping information without noting an error afaics.

Reply:
i saw that coming :grimacing: ... according to you what is the correct behaviour? add it directly to `decoder.Err` with some ad hoc message and abort? log it and move on?

Reply:
Good question. IMO the most consistent way would be to return an error and reject the event. Creating a custom decoding error and return sounds reasonable to me.

Reply:
done in 67e819b02bf41dddc5f5d4a1139f8061c8dedd7b